templates/zonefile.j2

```diff
@@ -9,51 +9,51 @@ $TTL 1W
 	IN	MX 10	smtp.{{ dns.clusterid }}.{{ dns.domain | lower }}.
 ;
 ; 
-ns1	IN	A	{{ helper.ipaddr }}
-smtp	IN	A	{{ helper.ipaddr }}
+ns1			IN	A	{{ helper.ipaddr }}
+smtp			IN	A	{{ helper.ipaddr }}
 ;
-{{ helper.name | lower }}	IN	A	{{ helper.ipaddr }}
+{{ helper.name | lower }}			IN	A	{{ helper.ipaddr }}
 ;
 ; The api points to the IP of your load balancer
-api		IN	A	{{ helper.ipaddr }}
-api-int		IN	A	{{ helper.ipaddr }}
+api			IN	A	{{ helper.ipaddr }}
+api-int			IN	A	{{ helper.ipaddr }}
 ;
 ; The wildcard also points to the load balancer
-*.apps		IN	A	{{ helper.ipaddr }}
+*.apps			IN	CNAME	{{ helper.name | lower }}.{{ dns.clusterid }}.{{ dns.domain | lower }}.
 ;
 {%if setup_registry %}
 ; Create entry for the local registry
-{{ registry_host }}	IN	A	{{ helper.ipaddr }}
+{{ registry_host.split('.')[0] }}		IN	A	{{ helper.ipaddr }}
 ;
 {% endif %}
 ; Create entry for the bootstrap host
-{{ bootstrap.name | lower }}	IN	A	{{ bootstrap.ipaddr }}
+{{ bootstrap.name.split('.')[0] }}		IN	A	{{ bootstrap.ipaddr }}
 ;
 ; Create entries for the master hosts
 {% for m in masters %}
-{{ m.name | lower }}		IN	A	{{ m.ipaddr }}
+{{ m.name.split('.')[0] }}		IN	A	{{ m.ipaddr }}
 {% endfor %}
 ;
 ; Create entries for the worker hosts
 {% for w in workers %}
-{{ w.name | lower }}		IN	A	{{ w.ipaddr }}
+{{ w.name.split('.')[0] }}		IN	A	{{ w.ipaddr }}
 {% endfor %}
 ;
 ; The ETCd cluster lives on the masters...so point these to the IP of the masters
 {% for m in masters %}
-etcd-{{ loop.index0 }}	IN	A	{{ m.ipaddr }}
+etcd-{{ loop.index0 }}			IN	A	{{ m.ipaddr }}
 {% endfor %}
 ;
 {% if other is defined %}
 ; Create entries for the other hosts
 {% for o in other %}
-{{ o.name }}		IN	A	{{ o.ipaddr }}
+{{ o.name }}				IN	A	{{ o.ipaddr }}
 {% endfor %}
 ;
 {% endif %}
 ; The SRV records are IMPORTANT....make sure you get these right...note the trailing dot at the end...
 {% for m in masters %}
-_etcd-server-ssl._tcp	IN	SRV	0 10 2380 etcd-{{ loop.index0 }}.{{ dns.clusterid }}.{{ dns.domain | lower }}.
+_etcd-server-ssl._tcp.{{ dns.clusterid }}.{{ dns.domain | lower }}.	86400	IN	SRV	0 10 2380 etcd-{{ loop.index0 }}.{{ dns.clusterid }}.{{ dns.domain | lower }}.
```

Comment:
Why is the FQDN needed here?

Reply:
Also why the TTL?

Reply:
According to the official Red Hat doc it should be like that.
https://docs.openshift.com/container-platform/4.3/installing/installing_ibm_power/installing-ibm-power.html#installation-dns-user-infra_installing-ibm-power

Reply:
Good catch! Didn't realize that. In that case, I want to keep it.